wolfcrypt/src/misc.c

```diff
@@ -182,9 +182,11 @@ STATIC WC_INLINE void XorWords(wolfssl_word* r, const wolfssl_word* a, word32 n)
 
 STATIC WC_INLINE void xorbuf(void* buf, const void* mask, word32 count)
 {
-    if (((wolfssl_word)buf | (wolfssl_word)mask | count) % WOLFSSL_WORD_SIZE == 0)
+    if (((wolfssl_word)((size_t)buf) | (wolfssl_word)((size_t)mask) | count) %
```

Comment:
Why is the double case with a size_t needed now?

Reply:
@toddouska : This fix resolves a warning with conversion of pointer from different sizes. With `NO_64BIT` defined the `wolfssl_word` is 32-bits and `void*` is still 64-bits. Specifically when I was testing with Ubuntu 64-bit VM on my Mac.

```
./wolfcrypt/src/misc.c: In function 'xorbuf':
./wolfcrypt/src/misc.c:185:10: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
     if (((wolfssl_word)(buf) | (wolfssl_word)(mask) | count) %
          ^
```

I have updated the code to this, which is a bit cleaner:

```
    if ((wolfssl_word)((size_t)buf | (size_t)mask | count) %
                                                       WOLFSSL_WORD_SIZE == 0) {
```

Please let me know if you've got a better idea for fixing this.

Reply:
Why are we defining NO_64BIT on a platform that has a 64bit type available?  Some people will want to force 32bit instructions, in which case the pointer being 64bits I'm not sure makes sense.  We've been testing (or should have been) -m32 on 64bit machines for several years w/o this problem.

Reply:
NO_64BIT in 64bit mode (instructions, pointers) doesn't make sense to me either.  NO_64BIT was originally for 32bit machines with a 64bit type that may result in slower multiplies than the native type (32bit) multiply.